up-l1/someip.adoc

```diff
@@ -0,0 +1,221 @@
+By using SOME/IP, automotive ECUs can communicate with each other to exchange information related to vehicle functions, such as sensor data, control commands, diagnostics, and software updates. It enables the integration and interoperability of different ECUs from various automotive suppliers within a vehicle's network architecture.
+Overall, SOME/IP plays a crucial role in facilitating efficient and standardized communication between ECUs in modern automotive systems.
+
+The following specification will elaborate on how uProtocol shall be mapped to SOME/IP protocol so that mechatronics applications and services can be connected to non-mechatronics and cloud based applications and services. 
+
+=== Open Source Libraries
+
+As of the time of writing, there are two open source projects that are available for SOME/IP:
+
+1. *https://github.com/COVESA/vsomeip[COVESA-vsomeip]:* C++ solution based on the original specifications developed by BMW prior to contribution to AUTOSAR
+2. *https://projects.eclipse.org/projects/automotive.sommr[Eclipse-SommR]:*  Rust implementation based on the latest AUTOSAR SOME/IP specifications. 
+
+NOTE: Given the latest version of SOME/IP specifications from AUTOSAR is not open source, we cannot share or implement uProtocol using Eclipse-SommR.
+
+
+== Transport Layer Translation
+
+=== UAttributes
+
+The following table highlights the mapping of uAttributes to SOME/IP fields that apply to all event types. For more information about uAttributes please refer to link:../basics/uri.adoc[URI].
+
+
+==== Uris
+
+link:../basics/uattributes.adoc[UAttributes Specification] explains that `source` attributes defines the address of whom sent the message, while `sink` defines the destination for the message in link:../basics/uri.adoc[`UUri`] format. `UUri` is translated to SOME/IP attributes per  <<sink-source-uri-mapping>> below.
+
+.Sink & Source URI Mapping
+[sink-source-uri-mapping]
+|===
+|*UUri Portion* |*SOME/IP Field* |*Description*
+|*`UAuthority::ip`* |IP address & port |IP address & port (destination
+endpoint) of the mDevice
+
+|*`UEntity::id`* |Service ID |Software entity ID that is placed in the upper 16
+bits of the SOME/IP Message ID header field
+
+|*`UResource::id`* |Method/Event ID |Identifier of the method/resource placed in the
+lower 15 bits of the SOME/IP Message ID header field
+
+|===
+
+
+==== UUIDs
+
+link:../basics/uuid.adoc[uProtocol UUID specifications] create a unique identifier for each message along with timestamp information. the ID is used for correlate between request and response as well. SOME/IP instead defines the ID as 16bit client ID + 16 bit session ID (that is incremented). When messages are converted SOME/IP to/from uProtocol, care must be taken to ensure that the SOME/IP ID and uProtocol UUID are properly mapped, especially when corelating a request to a response. 
+
+* Generated SOME/IP events *SHOULD* set the 16 bit client ID to 0 per SOME/IP specifications 4.1.2.4
```

Comment:
Maybe you should refer to `[PRS_SOMEIP_00925]`?

> • Set the Client ID to 0x00. Set the Session ID according to
[PRS_SOMEIP_00932], [PRS_SOMEIP_00933], and [PRS_SOMEIP_00521].
In case of active Session Handling the Session ID shall be incremented upon
each transmission.

Reply:
done